### PR TITLE
Recalculate table width when columns are shown/hidden

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Addresses #7 

This update will also emit **hidden** column widths in the `columnWidthChange` output emitter. This will allow consumers to retain user width settings even after they show/hide columns.

### CODE REVIEW

File: src/app/public/modules/grid/grid.component.ts [hunk 1]
```diff
@@ -539,6 +540,15 @@ export class SkyGridComponent implements AfterContentInit, OnChanges, OnDestroy
     this.startColumnWidth = column.width;
   }
 
+  private resetTableWidth() {
+    this.gridAdapter.setStyle(this.tableElementRef, 'width', `auto`);
+    this.ref.detectChanges();
+    this.tableWidth = this.tableElementRef.nativeElement.offsetWidth;
+    this.gridAdapter.setStyle(this.tableElementRef, 'width', `${this.tableWidth}px`);
+    this.tableWidth = this.tableElementRef.nativeElement.offsetWidth;
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.ts]: You have to set the `this.tableWidth` twice?

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: That was a oversight. Good catch! This has been fixed.

File: src/app/public/modules/grid/grid.component.spec.ts [hunk 2]
```diff
@@ -702,6 +768,9 @@ describe('Grid Component', () => {
         imports: [
           GridFixturesModule,
           SkyGridModule
+        ],
+        providers: [
+          SkyWindowRefService
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.spec.ts]: More for future reference, but you could provide `SkyWindowRefService` in `GridFixturesModule` instead, so you don't duplicate the code.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.spec.ts]: Fixed.

**Pull request merged**